neptune/internal/storage/storage_utils.py

```diff
@@ -95,8 +101,22 @@ def __repr__(self):
 
 def split_upload_files(upload_entries, max_package_size=1 * 1024 * 1024, max_files=500):
     current_package = UploadPackage()
+    walked_entries = set()
     for entry in upload_entries:
-        size = 0 if os.path.isdir(entry.source_path) else os.path.getsize(entry.source_path)
+        if os.path.isdir(entry.source_path):
+            for root, _, files in os.walk(entry.source_path):
+                path_relative_to_entry_source = os.path.relpath(root, entry.source_path)
+                if path_relative_to_entry_source == ".":
+                    target_root = entry.target_path
+                else:
+                    target_root = entry.target_path + "/" + path_relative_to_entry_source
+                for file in files:
+                    walked_entries.add(UploadEntry(root + "/" + file, target_root + "/" + file))
```

Comment:
constructing paths like this is dangerous, use os.path.join. Even so, this should be tested on windows

Reply:
Also, please use some name other than `file`. I think it'd bad form in Python, since this is shadows a built-in.